app/pages/settings/profile.jsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+import apiClient from 'panoptes-client/lib/api-client';
+import ImageSelector from '../../components/image-selector';
+import putFile from '../../lib/put-file';
+
+const MAX_AVATAR_SIZE = 65536;
+const MAX_HEADER_SIZE = 256000;
+
+class CustomiseProfile extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      avatar: {},
+      avatarError: null,
+      profile_header: {},
+      profile_headerError: null
+    };
+    this.getUserResource('avatar');
+    this.getUserResource('profile_header');
+  }
+
+  getUserResource(type) {
+    return this.props.user.get(type, {})
+    .then(([resource]) => {
+      this.setState({ [type]: resource });
+      return resource;
+    });
+  }
+
+  handleMediaChange(type, file) {
+    const { user } = this.props;
+    const errorProp = `${type}Error`;
+    this.setState({ [errorProp]: null });
+    apiClient.post(user._getURL(type), { media: { content_type: file.type }})
+    .then(([resource]) => {
```

Comment:
`resource.src` here is a signed S3 URL, for uploading the file, so we can't store this resource in component state.